client/components/OperationList.js

```diff
@@ -165,6 +165,142 @@ function ComputeAttachmentLink(op) {
     return `operations/${op.id}/`+file;
 }
 
+
+
+class LabelComponent extends React.Component {
+    constructor(props) {
+        super(props);
+        this.state = {
+            editMode: false            
+        };
+    }
+
+    dom() {
+        return this.refs.customlabel.getDOMNode();
+    }
+
+    switchToEditMode() {
+        this.setState({ editMode: true }, function() {
+            this.dom().focus();
+            //We set the cursor at the end
+            try {
+                this.dom().selectionStart = this.props.defaultValue().length;
+            }
+            catch (err){
+               this.dom().selectionStart = 0;
+           }
+        });
+    }
+    switchToStaticMode() {
+        this.setState({ editMode: false });
+    }
+
+    onBlur() {
+        let customLabel = this.dom().value;
+        Actions.SetCustomLabel(this.props.operation, customLabel);
+        // Be optimistic
+        this.props.operation.customLabel = customLabel;
+        this.switchToStaticMode();
+    }
+    
+    onKeyUp(e) {
+        if (e.keyCode == 13) {
+            this.onBlur();
+        }        
+    }
+    
+    render() {
+        if (!this.state.editMode){
+            return (
+                <button
+                    className="form-control text-left btn-transparent"
+                    id={this.props.operation.id}
+                    onClick={this.switchToEditMode.bind(this)}>
```

Comment:
nit: for the previous 3 lines (that is, inside the <button> tag), can you half-indent (2 spaces), please? It makes a clear visual difference between the attributes and the content.